src/conversion/conversion.js

```diff
@@ -149,7 +183,18 @@ export default class Conversion {
 	 * @returns {module:engine/conversion/downcasthelpers~DowncastHelpers|module:engine/conversion/upcasthelpers~UpcastHelpers}
 	 */
 	for( groupName ) {
-		return this._getConversionHelpers( groupName );
+		if ( !this._groups.has( groupName ) ) {
+			/**
+			 * Trying to add a converter to an unknown dispatchers group.
+			 *
+			 * @error conversion-for-unknown-group
+			 */
+			throw new CKEditorError( 'conversion-for-unknown-group: Trying to add a converter to an unknown dispatchers group.' );
+		}
+
+		const ConversionHelper = this._helpers.get( groupName );
+
+		return new ConversionHelper( this._groups.get( groupName ) );
```

Comment:
Is it efficient to create a new instance whenever the converter is used? Maybe we could store instances of helpers?